src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -214,25 +194,20 @@ export function XYChart({
     yRight: true,
   };
 
-  const labelsOrientation = args.labelsOrientation || {
-    x: 0,
-    yLeft: 0,
-    yRight: 0,
-  };
+  const labelsOrientation = args.labelsOrientation || { x: 0, yLeft: 0, yRight: 0 };
 
-  const filteredBarLayers = filteredLayers.filter((layer) => layer.seriesType.includes('bar'));
+  const filteredBarLayers = dataLayers.filter((layer) => layer.seriesType.includes('bar'));
 
   const chartHasMoreThanOneBarSeries =
     filteredBarLayers.length > 1 ||
     filteredBarLayers.some((layer) => layer.accessors.length > 1) ||
-    filteredBarLayers.some((layer) => layer.splitAccessor);
+    filteredBarLayers.some((layer) => isDataLayer(layer) && layer.splitAccessor);
 
-  const isTimeViz = Boolean(filteredLayers.every((l) => l.xScaleType === 'time'));
-  const isHistogramViz = filteredLayers.every((l) => l.isHistogram);
+  const isTimeViz = Boolean(filteredLayers.every((l) => isDataLayer(l) && l.xScaleType === 'time'));
```

Comment:
This is changing the logic - if there is a non-data layer (e.g. a reference line layer), it will make `isTimeViz` false even if all the data lyers satisfy the `xScaleType === 'time'` condition. Same with `isHistogramViz`

Reply:
Were `filterLayers` all the time `dataLayers` only? If not, this logic was working like that all the time.
If the layer was `referenceLineLayer`, layer.xScaleType was undefined and was returning false. So, my code just highlighted the existing problem.

I'll change from `filteredLayers` to `dataLayers`.

Reply:
Done.